e2e/tests/datasets/export-to-notebook.spec.ts

```diff
@@ -1,61 +1,277 @@
+    dataPage = await cohortActionsPage.clickDataTab();
+
+    // Verify dataset exists.
+    await dataPage.openDatasetsSubtab();
+    const resourceCard = new DataResourceCard(page);
+    const dataSetExists = await resourceCard.cardExists(datasetName, ResourceCard.Dataset);
+    expect(dataSetExists).toBe(true);
+
+    // Find Dataset card. Open Edit Dataset page.
+    const datasetCard = await resourceCard.findCard(datasetName, ResourceCard.Dataset);
+    await datasetCard.selectSnowmanMenu(MenuOption.Edit, { waitForNav: true });
+
+    const datasetEditPage = new DatasetEditPage(page);
+    await datasetEditPage.waitForLoad();
+
+    saveModal = await datasetEditPage.clickAnalyzeButton();
+
+    // Leave Dataset name unchanged.
+    expect(await saveModal.getNameTextbox().getValue()).toEqual(datasetName);
+
+    // Select previously created (existing) notebook to export to.
+    await saveModal.saveDataset(
+      {
+        exportToNotebook: true,
+        notebookName,
+        datasetName
+      },
+      true
+    );
+
+    const notebookPreviewPage = new NotebookPreviewPage(page);
+    await notebookPreviewPage.waitForLoad();
+
+    // Open notebook in Edit mode.
+    const notebookPage = await notebookPreviewPage.openEditMode(notebookName);
+
+    // Run dataset in notebook cell #1.
+    await notebookPage.runCodeCell(1);
+
+    // Checkout code cell output.
+    const cell1 = notebookPage.findCell(1);
+    const cell1Output = await cell1.findOutputElementHandle();
+
+    // Cell output format should be html table.
+    const outputClassName = await getPropValue<string>(cell1Output, 'className');
+    expect(outputClassName).toContain('rendered_html');
+
+    // Verify workspace name in notebook page.
+    const workspaceLink = await notebookPage.getWorkspaceLink().asElementHandle();
+    expect(await getPropValue<string>(workspaceLink, 'textContent')).toEqual(workspace);
+
+    // Navigate to Workspace Data page.
+    await notebookPage.goDataPage();
+
+    // Delete the cohort.
+    await deleteCohort(cohortName, datasetName);
+
+    // Dataset is no longer exists.
+    await dataPage.openDatasetsSubtab();
+    expect(await resourceCard.cardExists(datasetName, ResourceCard.Dataset)).toBe(false);
+
+    // Delete the notebook.
+    await dataPage.openAnalysisPage();
+    const analysisPage = new WorkspaceAnalysisPage(page);
+    await analysisPage.waitForLoad();
+    await analysisPage.deleteResource(notebookName, ResourceCard.Notebook);
+
+    // Finally, delete the workspace.
+    await dataPage.deleteWorkspace();
+  });
 
   /**
    * Test:
-   * - Create dataset.
-   * - Export dataset to notebook thru snowman menu.
+   * - Create dataset and export to notebook. Start the notebook and run the dataset code.
    */
-  test('Create Jupyter notebook for Python programming language from existing dataset', async () => {
+  test.each(ProgrammingLanguage)('Export to %s kernel Jupyter notebook when creating dataset', async (each) => {
```

Comment:
`test.each` makes this test data-driven. Test run for Python and R language independently.